### PR TITLE
Adding zpool API to retrieve allocated and total size of a zpool

### PR DESCRIPTION
**Pull request by raj-kumar-soni**

### CODE REVIEW

File: include/sys/zfs_ioctl.h [hunk 1]
```diff
@@ -636,7 +648,7 @@ extern void zfs_unmount_snap(const char *);
 extern void zfs_destroy_unmount_origin(const char *);
 extern int getzfsvfs_impl(struct objset *, struct zfsvfs **);
 extern int getzfsvfs(const char *, struct zfsvfs **);
-extern void latency_stats(uint64_t* histo, unsigned int buckets,
+extern void latency_stats(uint64_t * histo, unsigned int buckets,
```
Comment (datacore-rm) [inline include/sys/zfs_ioctl.h:651]: additional space is there after *

File: include/sys/zfs_ioctl.h [hunk 2]
```diff
@@ -636,7 +648,7 @@ extern void zfs_unmount_snap(const char *);
...
     stat_pair* lat);
```
Comment (datacore-rm) [inline include/sys/zfs_ioctl.h:653]: additional space is there after * in both lines

File: module/os/windows/zfs/zvol_os.c [hunk 3]
```diff
@@ -1035,3 +1035,87 @@ zvol_fini(void)
...
+    PIO_STACK_LOCATION IrpSp)
+{
+	if (IrpSp->Parameters.DeviceIoControl.OutputBufferLength <
+	    sizeof (zpool_size_stats)) {
+		Irp->IoStatus.Information = sizeof (zpool_size_stats);
+		return (STATUS_BUFFER_TOO_SMALL);
+	}
+
+	zpool_size_stats* zpool_stats =
+	    (zpool_size_stats*)Irp->AssociatedIrp.SystemBuffer;
```
Comment (datacore-rm) [inline module/os/windows/zfs/zvol_os.c]: cstyle for pointer

File: module/os/windows/zfs/zvol_os.c [hunk 4]
```diff
@@ -1035,3 +1035,87 @@ zvol_fini(void)
 	zvol_fini_impl();
 	taskq_destroy(zvol_taskq);
 }
+
+void
+get_zpool_name(PDEVICE_OBJECT DeviceObject, char *zpool_name,
+    size_t size) {
```
Comment (datacore-rm) [inline module/os/windows/zfs/zvol_os.c]: If this function is not called then pls remove.

File: module/os/windows/zfs/zfs_vnops_windows.c [hunk 5]
```diff
@@ -4762,6 +4774,11 @@ _Function_class_(DRIVER_DISPATCH)
 			Status = ioctl_disk_get_drive_geometry(DeviceObject,
 			    Irp, IrpSp);
 			break;
+		case ZPOOL_GET_SIZE_STATS:
```
Comment (datacore-rm) [inline module/os/windows/zfs/zfs_vnops_windows.c]: I think this ioctl is sent from pool driver to \\Device\\ZFSCTL which is already handled above in ioctlDispatcher() and not required to be handled here?

File: module/os/windows/zfs/zvol_os.c (hunk 3 above)
Comment (raj-kumar-soni) [reply module/os/windows/zfs/zvol_os.c]: done

File: module/os/windows/zfs/zvol_os.c (hunk 4 above)
Comment (raj-kumar-soni) [reply module/os/windows/zfs/zvol_os.c]: done

File: include/sys/zfs_ioctl.h (hunk 2 above)
Comment (raj-kumar-soni) [reply include/sys/zfs_ioctl.h:653]: done

File: module/os/windows/zfs/zfs_vnops_windows.c (hunk 5 above)
Comment (raj-kumar-soni) [reply module/os/windows/zfs/zfs_vnops_windows.c]: Removed case from diskdispatcher

**Review by datacore-rm — APPROVED** [review]

**Pull request merged**